sweepbatcher/sweep_batch.go

```diff
@@ -561,46 +520,151 @@ func (b *batch) addSweep(ctx context.Context, sweep *sweep) (bool, error) {
 			int32(math.Abs(float64(sweep.timeout - s.timeout)))
 
 		if timeoutDistance > b.cfg.maxTimeoutDistance {
-			b.Infof("too long timeout distance between the "+
-				"batch and sweep %x: %d > %d",
+			return false, fmt.Errorf("too long timeout distance "+
+				"between the batch and sweep %x: %d > %d",
 				sweep.swapHash[:6], timeoutDistance,
 				b.cfg.maxTimeoutDistance)
+		}
+	}
+
+	// Everything is ok, the sweep can be added to the batch.
+	return false, nil
+}
+
+// addSweeps tries to add sweeps to the batch. If this is the first sweep being
+// added to the batch then it also sets the primary sweep ID. It returns if the
+// sweeps were accepted to the batch.
+func (b *batch) addSweeps(ctx context.Context, sweeps []*sweep) (bool, error) {
+	done, err := b.scheduleNextCall()
+	defer done()
+	if err != nil {
+		return false, err
+	}
+
+	// Track if there is an existing sweep and a new sweep among the sweeps.
+	var numExisting, numNew int
+	for _, s := range sweeps {
+		existing, err := b.checkSweepToAdd(ctx, s)
+		if err != nil {
+			b.Infof("Failed to add sweep %v to batch %d: %v",
+				s.outpoint, b.id, err)
+
+			return false, nil
+		}
+		if existing {
+			numExisting++
+		} else {
+			numNew++
+		}
+	}
+
+	// Make sure the whole group is either new or existing. If this is not
+	// the case, this might be a bug, so print a warning.
+	if numExisting > 0 && numNew > 0 {
```

Comment:
nit: we could also just check `countExisting != len(sweeps)`.

Reply:
We shouldn't return an error, if all the sweeps added are new. If all sweeps are new, countExisting=0 (and `numNew=len(sweeps)`).